File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/collection_join_provider.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

public class collection_join_provider : IDisposable {
  private HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal collection_join_provider(IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(collection_join_provider obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~collection_join_provider() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_collection_join_provider(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
    }
  }

  public collection_join_provider() : this(gnsdk_csharp_marshalPINVOKE.new_collection_join_provider(), true) {
  }

  public GnPlaylistCollection get_data(uint pos) {
    GnPlaylistCollection ret = new GnPlaylistCollection(gnsdk_csharp_marshalPINVOKE.collection_join_provider_get_data(swigCPtr, pos), true);
    return ret;
  }

  public uint count() {
    uint ret = gnsdk_csharp_marshalPINVOKE.collection_join_provider_count(swigCPtr);
    return ret;
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/collection_join_provider.java

package com.gracenote.gnsdk;

public class collection_join_provider {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected collection_join_provider(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(collection_join_provider obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_collection_join_provider(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public collection_join_provider() {
    this(gnsdk_javaJNI.new_collection_join_provider(), true);
  }

  public GnPlaylistCollection getData(long pos) {
    return new GnPlaylistCollection(gnsdk_javaJNI.collection_join_provider_getData(swigCPtr, this, pos), true);
  }

  public long count() {
    return gnsdk_javaJNI.collection_join_provider_count(swigCPtr, this);
  }

}
